dali/operators/random/uniform.cc

```diff
@@ -17,26 +17,63 @@
 
 namespace dali {
 
-void Uniform::RunImpl(HostWorkspace &ws) {
+void Uniform::AssignRange(HostWorkspace &ws) {
+  auto &output = ws.OutputRef<CPUBackend>(0);
+  auto dist = std::uniform_real_distribution<float>(range_[0], range_[1]);
+  for (int i = 0; i < batch_size_; ++i) {
+    auto *sample_data = output[i].mutable_data<float>();
+    auto sample_len = output[i].size();
+    for (int k = 0; k < sample_len; ++k) {
+      sample_data[k] = dist(rng_);
+    }
+  }
+}
+
+
+void Uniform::AssignSet(HostWorkspace &ws) {
   auto &output = ws.OutputRef<CPUBackend>(0);
+  auto dist = std::uniform_int_distribution<int>(0, set_.size() - 1);
   for (int i = 0; i < batch_size_; ++i) {
     auto *sample_data = output[i].mutable_data<float>();
     auto sample_len = output[i].size();
     for (int k = 0; k < sample_len; ++k) {
-      sample_data[k] = dis_(rng_);
+      sample_data[k] = set_[dist(rng_)];
     }
   }
 }
 
+
+void Uniform::RunImpl(HostWorkspace &ws) {
+  if (discrete_mode_) {
+    AssignSet(ws);
+  } else {
+    AssignRange(ws);
+  }
+}
+
 DALI_REGISTER_OPERATOR(Uniform, Uniform, CPU);
 
 DALI_SCHEMA(Uniform)
-  .DocStr("Produces random numbers following a uniform distribution.")
+  .DocStr(R"code(Produces random numbers following an uniform distribution.
+
+Both continuous and discrete uniform distributions can be defined by defining
+a continuous ``range`` or a discrete ``set`` of values, respectively.
```

Comment:
```suggestion
a continuous ``range`` or a discrete list of ``values``, respectively.
```
A set doesn't  allow repetitions - and the user may want them.